Module 4 Block 4 : OOP school library: preserve data

aceDavon opened the pull request:
## In this PR, we have implemented the following tasks: :writing_hand: 

- [ ] Store data in files.
- [ ] Read data from local files on load of application
- [ ] Run a program using the command line.

betses committed:
Add preserv data functions

aceDavon committed:
Write to the local data file

Write user data to the individual local json files, read data from the
files on subsequent loads

aceDavon committed:
Edit the Rubocop.yml file to allow more lines

edit the rubocop yaml file to allow for more lines per method definition

youmari commented on database.rb:
```diff
@@ -0,0 +1,15 @@
+    File.write('./data/people.json', JSON.generate(people))
+  end
+
+  def save_rentals(rentals)
+    File.write('./data/rentals.json', JSON.generate(rentals))
+  end
+end
```
- [ ] while creating a rental I got the following error please make sure to fix it in order for your app will work properly 
![Screen Shot 2022-11-08 at 15 15 10](https://user-images.githubusercontent.com/61790587/200588116-91ddd915-cb35-4957-907a-ad009304285b.png)

youmari reviewed (changes requested):
Hi @aceDavon @betses , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

aceDavon committed:
Fix bugs with create rentals

fix the error with create rentals, Assign the rental class to the rental
 array per person instance passed thru the create_rental method of the
 app

Ugboaja-Uchechi reviewed (approved):
Hi @aceDavon, 

## Status: APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
